### PR TITLE
ath79: Add support for WNDRMAC, WNDRMACv2

### PR DESCRIPTION
**Pull request by cybik**
Adding router support for two devices
* The Netgear WNDRMAC is a variant of the WNDR3700v1
* The Netgear WNDRMACv2 is a variant of the Netgear WNDR3800

Specification: see as above

(wip)

### CODE REVIEW

File: target/linux/ath79/image/generic.mk [hunk 1]
```diff
@@ -899,10 +899,28 @@ define Device/netgear_wndr3800
   NETGEAR_BOARD_ID := WNDR3800
   NETGEAR_HW_ID := 29763654+16+128
   IMAGE_SIZE := 15872k
-  SUPPORTED_DEVICES += wndr3700
+  SUPPORTED_DEVICES += wndr3800
```
Comment (adrianschmutzler) [inline target/linux/ath79/image/generic.mk:902]: I don't think that's correct. Despite, if it's valid, it should go into a separate commit.

Comment (cybik) [reply target/linux/ath79/image/generic.mk:902]: Fair. I'll remove that.

**Comment by adrianschmutzler** [comment]
Your commit has formal issues. Please have a look at https://openwrt.org/submitting-patches

From first look, you need to
- use proper commit title
- add specifications/flashing instructions
- maybe split commits depending on how different the device specs are

File: target/linux/ath79/image/generic.mk [hunk 2]
```diff
@@ -903,6 +903,24 @@ define Device/netgear_wndr3800
 endef
 TARGET_DEVICES += netgear_wndr3800
 
+define Device/netgear_wndrmac
+  $(Device/netgear_wndr3700-v2)
+  DEVICE_MODEL := WNDRMAC
+  DEVICE_VARIANT := v1
+  NETGEAR_BOARD_ID := WNDRMAC
+  SUPPORTED_DEVICES += wndrmac
```
Comment (adrianschmutzler) [inline target/linux/ath79/image/generic.mk:911]: To me, it looks like this one should be wndr3700 as well.

File: target/linux/ath79/image/generic.mk [hunk 3]
```diff
@@ -903,6 +903,24 @@ define Device/netgear_wndr3800
...
+endef
+TARGET_DEVICES += netgear_wndrmac
+
+define Device/netgear_wndrmacv2
+  $(Device/netgear_wndr3800)
+  DEVICE_MODEL := WNDRMAC
+  DEVICE_VARIANT := v2
+  NETGEAR_BOARD_ID := WNDRMACv2
+  SUPPORTED_DEVICES += wndrmacv2
```
Comment (adrianschmutzler) [inline target/linux/ath79/image/generic.mk:920]: wndr3700

File: target/linux/ath79/image/generic.mk [hunk 4]
```diff
@@ -903,6 +903,24 @@ define Device/netgear_wndr3800
 endef
 TARGET_DEVICES += netgear_wndr3800
 
+define Device/netgear_wndrmac
```
Comment (adrianschmutzler) [inline target/linux/ath79/image/generic.mk:906]: netgear_wndrmac-v1

File: target/linux/ath79/image/generic.mk [hunk 5]
```diff
@@ -903,6 +903,24 @@ define Device/netgear_wndr3800
...
+  $(Device/netgear_wndr3700-v2)
```
Comment (adrianschmutzler) [inline target/linux/ath79/image/generic.mk:907]: Please use $(Device/netgear_wndr3x00) here.

File: target/linux/ath79/image/generic.mk [hunk 6]
```diff
@@ -903,6 +903,24 @@ define Device/netgear_wndr3800
...
+  DEVICE_MODEL := WNDRMAC
+  DEVICE_VARIANT := v1
+  NETGEAR_BOARD_ID := WNDRMAC
+  SUPPORTED_DEVICES += wndrmac
+endef
+TARGET_DEVICES += netgear_wndrmac
+
+define Device/netgear_wndrmacv2
```
Comment (adrianschmutzler) [inline target/linux/ath79/image/generic.mk:915]: netgear_wndrmac-v2

File: target/linux/ath79/image/generic.mk [hunk 7]
```diff
@@ -903,6 +903,24 @@ define Device/netgear_wndr3800
...
+  $(Device/netgear_wndr3800)
```
Comment (adrianschmutzler) [inline target/linux/ath79/image/generic.mk:916]: $(Device/netgear_wndr3x00)

File: target/linux/ath79/image/generic.mk [hunk 8]
```diff
@@ -903,6 +903,24 @@ define Device/netgear_wndr3800
...
+  DEVICE_MODEL := WNDRMAC
+  DEVICE_VARIANT := v2
+  NETGEAR_BOARD_ID := WNDRMACv2
+  SUPPORTED_DEVICES += wndrmacv2
+endef
+TARGET_DEVICES += netgear_wndrmacv2
+
 define Device/netgear_wndr3800ch
```
Comment (adrianschmutzler) [inline target/linux/ath79/image/generic.mk:924]: Please care about alphabetic sorting, wndrmac should after after wndr with numbers.

File: target/linux/ath79/generic/base-files/etc/board.d/02_network [hunk 9]
```diff
@@ -199,7 +199,9 @@ ath79_setup_interfaces()
 		;;
 	netgear,wndr3700|\
 	netgear,wndr3700-v2|\
+	netgear,wndrmac|\
 	netgear,wndr3800|\
+	netgear,wndrmacv2|\
```
Comment (adrianschmutzler) [inline target/linux/ath79/generic/base-files/etc/board.d/02_network:204]: Alphabetic sorting.

File: target/linux/ath79/image/generic.mk (hunk 5 above)
Comment (cybik) [reply target/linux/ath79/image/generic.mk:907]: If I pull `$(Device/netgear_wndr3x00)` here I'd have to redefine everything `$(Device/netgear_wndr3700-v2)` does, which I don't want to do. I'm technically *relying* on the predefines.

**Comment by adrianschmutzler** [comment]
This PR is incomplete, it entirely lacks the DTS files. Therefore, it's not even compile-tested by the author. 

Anyway, without somebody to test on the devices, this won't be merged. However, if there are testers, it should be relatively straightforward to create support, as the devices seem to be stupid clones.

File: target/linux/ath79/image/generic.mk (hunk 7 above)
Comment (cybik) [reply target/linux/ath79/image/generic.mk:916]: Same as above. `netgear_wndr3800` has presets I wish to inherit.

**Comment by adrianschmutzler** [comment]
@cybik Note that the relevant changes are the ones in the commit, not in the PR.

File: target/linux/ath79/image/generic.mk (hunk 7 above)
Comment (cybik) [reply target/linux/ath79/image/generic.mk:916]: (Will do as is being requested)

File: target/linux/ath79/image/generic.mk (hunk 5 above)
Comment (cybik) [reply target/linux/ath79/image/generic.mk:907]: (Will do as is being requested)

**Comment by adrianschmutzler** [comment]
@cybik You seem to have some problems to understand how ath79 works, as some things are missing.  Maybe have a look at the commit where WNDR3800CH was added, this should serve as a blueprint: https://github.com/openwrt/openwrt/commit/00ed75f47f4726dff17c6d63d6847bc4f7a74970

However, as stated earlier, we won't accept a device without runtesting. Therefore, it might be better to create two commits, one for each device, as then you won't have to wait for both to be tested.

**Comment by cybik** [comment]
I'll be splitting this into two then. And I've no issues considering this (these?) PR/s as WIP

File: target/linux/ath79/image/generic.mk (hunk 5 above)
Comment (adrianschmutzler) [reply target/linux/ath79/image/generic.mk:907]: This is essentially one of the lessons ar71xx has told: As you see with wndr3xxx/wndrmac devices in ar71xx, you have to go through up to four definitions for some devices to really find all of the variables set. That's quite inconvenient and misleading.
In ath79, in contrast, devices are not derived from each other, but just from specifically created shared nodes. While this might cause some lines to be duplicated, it's actually much easier to read and understand.

**Comment by cybik** [comment]
Split into two PRs
https://github.com/openwrt/openwrt/pull/3012
https://github.com/openwrt/openwrt/pull/3011
